rust/processor/src/processors/parquet_processors/parquet_objects_processor.rs

```diff
@@ -0,0 +1,243 @@
+                            concurrent_fungible_asset_supply: None,
+                            concurrent_fungible_asset_balance: None,
+                            token_identifier: None,
+                        });
+                    }
+                }
+            }
+
+            // Second pass to construct the object data
+            for (index, wsc) in changes.iter().enumerate() {
+                let index: i64 = index as i64;
+                match wsc.change.as_ref().unwrap() {
+                    Change::WriteResource(inner) => {
+                        if let Some((object, current_object)) = &Object::from_write_resource(
+                            inner,
+                            txn_version,
+                            index,
+                            &object_metadata_helper,
+                            block_timestamp,
+                        )
+                        .unwrap()
+                        {
+                            all_objects.push(object.clone());
+                            transaction_version_to_struct_count
+                                .entry(txn_version)
+                                .and_modify(|e| *e += 1)
+                                .or_insert(1);
+                            all_current_objects
+                                .insert(object.object_address.clone(), current_object.clone());
+                        }
+                    },
+                    Change::DeleteResource(inner) => {
+                        // Passing all_current_objects into the function so that we can get the owner of the deleted
+                        // resource if it was handled in the same batch
+                        if let Some((object, current_object)) = Object::from_delete_resource(
+                            inner,
+                            txn_version,
+                            index,
+                            &all_current_objects,
+                            &mut conn,
+                            query_retries,
+                            query_retry_delay_ms,
+                            block_timestamp,
+                        )
+                        .await
+                        .unwrap()
+                        {
+                            all_objects.push(object.clone());
+                            transaction_version_to_struct_count
+                                .entry(txn_version)
+                                .and_modify(|e| *e += 1)
+                                .or_insert(1);
+                            all_current_objects
+                                .insert(object.object_address.clone(), current_object.clone());
+                        }
+                    },
+                    _ => {},
+                };
+            }
+        }
+
+        let objects_parquet_data = ParquetDataGeneric {
+            data: all_objects,
+            transaction_version_to_struct_count: transaction_version_to_struct_count.clone(),
```

Comment:
why cloning the map? we only care about the current state? it reads to me that, transaction 0 with map(0), transaction 1 with map(0, 1), transaction 2, with map(0, 1, 2).

Reply:
it's for the batch of transactions

Reply:
yea makes sense